sdk/typescript/src/types/framework.ts

```diff
@@ -57,18 +62,113 @@ export class Coin {
     };
   }
 
-  static getBalance(
-    data: GetObjectDataResponse | SuiMoveObject
-  ): BN | undefined {
+  public static getID(obj: ObjectData): ObjectId {
+    if (isSuiMoveObject(obj)) {
+      return obj.fields.id.id;
+    }
+    return getObjectId(obj);
+  }
+
+  /**
+   * Convenience method for select coin objects that has a balance greater than or equal to `amount`
+   *
+   * @param amount coin balance
+   * @param exclude object ids of the coins to exclude
+   * @return a list of coin objects that has balance greater than `amount` in an ascending order
+   */
+  static selectCoinsWithBalanceGreaterThanOrEqual(
```

Comment:
I am confused about the use case of this one, iirc we only read res[0] here, so maybe just returning the first one?